classes/class4/homework-soln/routes/index.js

```diff
@@ -0,0 +1,165 @@
+/* CONNECT TO MONGOOSE */
+var mongoose = require('mongoose');
+mongoose.connect(process.env.MONGOURI || 'mongodb://localhost/test');
+
+
+/* DEFINE SCHEMAS */
+var ingredientSchema = mongoose.Schema({
+	name: String,
+	price: Number,
+	inStock: Boolean
+});
+var Ingredient = mongoose.model('Ingredient', ingredientSchema);
+
+var orderSchema = mongoose.Schema({
+	customer: String,
+	ingredients: Array,
+	ingredientStr: String
+});
+var Order = mongoose.model('Order', orderSchema);
+
+
+/* DEFINE ROUTE CALLBACKS */
+var routes = {};
+
+routes.ingredients = function(req, res) {
+// query for in stock & out of stock ingredients separately and render
+	Ingredient.find({'inStock':true}, function(err, inData) {
```

Comment:
You might consider doing 1 query to get all ingredients then separating inStock/outOfStock in the callback. It would likely more performant at scale.